pytensor/tensor/functional.py

```diff
@@ -0,0 +1,117 @@
+from typing import Callable, Optional
+
+from pytensor.graph import vectorize_graph
+from pytensor.tensor import TensorVariable
+from pytensor.tensor.utils import _parse_gufunc_signature
+
+
+def vectorize(func: Callable, signature: Optional[str] = None) -> Callable:
+    """Equivalent to numpy.vectorize.
+
+    If you have an existing graph you wish to vectorize, use `pytensor.graph.replace.vectorize_graph`
+
+    Parameters
+    ----------
+    func: Callable
+        Function that returns PyTensor varibales from TensorVariable inputs.
+    signature: str, optional
+        String that specifies the number and dimension of inputs and outputs in the core case.
```

Comment:
I think this could be a bit better explained.

Reply:
I think people will be familiar with how it works in numpy, and if not they can check it. We just borrowed their API